app/src/main/java/cm/aptoide/pt/editorialList/EditorialListManager.java

```diff
@@ -1,50 +1,59 @@
 package cm.aptoide.pt.editorialList;
 
+import cm.aptoide.pt.AppCoinsManager;
 import cm.aptoide.pt.reactions.ReactionsManager;
 import cm.aptoide.pt.reactions.network.LoadReactionModel;
 import cm.aptoide.pt.reactions.network.ReactionsResponse;
+import hu.akarnokd.rxjava.interop.RxJavaInterop;
 import java.util.List;
 import rx.Single;
 
 public class EditorialListManager {
 
-  private final EditorialListRepository editorialListRepository;
+  private final EditorialCardListRepository editorialCardListRepository;
   private final ReactionsManager reactionsManager;
+  private final AppCoinsManager appCoinsManager;
 
-  public EditorialListManager(EditorialListRepository editorialListRepository,
-      ReactionsManager reactionsManager) {
-    this.editorialListRepository = editorialListRepository;
+  public EditorialListManager(EditorialCardListRepository editorialCardListRepository,
+      ReactionsManager reactionsManager, AppCoinsManager appCoinsManager) {
+    this.editorialCardListRepository = editorialCardListRepository;
     this.reactionsManager = reactionsManager;
+    this.appCoinsManager = appCoinsManager;
   }
 
-  Single<EditorialListViewModel> loadEditorialListViewModel(boolean loadMore,
-      boolean invalidateCache) {
+  Single<EditorialListModel> loadEditorialListModel(boolean loadMore, boolean invalidateCache) {
     if (loadMore) {
-      return loadMoreCurationCards();
+      return loadMoreCurationCards().flatMap(
+          editorialCardListModel -> RxJavaInterop.toV1Single(appCoinsManager.getBonusAppc())
+              .map(bonusAppcModel -> new EditorialListModel(editorialCardListModel,
+                  bonusAppcModel)));
     } else {
-      return editorialListRepository.loadEditorialListViewModel(invalidateCache);
+      return editorialCardListRepository.loadEditorialCardListModel(invalidateCache)
+          .flatMap(
+              editorialCardListModel -> RxJavaInterop.toV1Single(appCoinsManager.getBonusAppc())
+                  .map(bonusAppcModel -> new EditorialListModel(editorialCardListModel,
```

Comment:
Somewhat duplication logic, we can still reduce the number of lines here with a method extraction.